plugins/rule_engines/irods_rule_language/src/configuration.cpp

```diff
@@ -594,6 +636,32 @@ int readRuleStructAndRuleSetFromFile( const char *ruleBaseName, const char *rule
     freeRErrorContent( &errmsgBuf );
     return res;
 }
+int readRuleStructAndRuleSetFromBuffer(const char* ruleBaseName, char* ruleBase)
+{
+    int errloc;
+    rError_t errmsgBuf;
+    errmsgBuf.errMsg = NULL;
+    errmsgBuf.len = 0;
+
+    char* buf = (char*) malloc(ERR_MSG_LEN * 1024 * sizeof(char));
+    int res = 0;
+    if ((res = readRuleSetFromBuffer(ruleBaseName,
+                                     ruleBase,
+                                     ruleEngineConfig.coreRuleSet,
+                                     ruleEngineConfig.coreFuncDescIndex,
+                                     &errloc,
+                                     &errmsgBuf,
+                                     ruleEngineConfig.coreRegion)) == 0)
+    {
+    }
+    else {
+        errMsgToString(&errmsgBuf, buf, ERR_MSG_LEN * 1024);
+        rodsLog(LOG_ERROR, "%s", buf);
```

Comment:
Let's use the new logger instead of `rodsLog`. `rodsLog` is going to be deprecated soon.